toolchain/check/cpp/thunk.cpp

```diff
@@ -243,8 +262,11 @@ static auto BuildThunkParameterTypes(clang::ASTContext& ast_context,
         GetNonnullType(ast_context, callee_info.implicit_this_type));
   }
 
-  for (const clang::ParmVarDecl* callee_param :
-       callee_info.decl->parameters()) {
+  for (auto [i, callee_param] :
+       llvm::enumerate(callee_info.decl->parameters())) {
+    if (static_cast<int>(i) >= callee_info.num_params) {
+      break;
+    }
```

Comment:
Just suggesting the different approach, copying from the earlier state
```suggestion
  for (const clang::ParmVarDecl* callee_param :
       callee_info.decl->parameters().take_front(callee_info.num_params)) {
```